src/arch/xtensa/include/arch/drivers/interrupt.h

```diff
@@ -41,6 +36,14 @@ static inline void arch_interrupt_unregister(int irq)
 #define arch_interrupt_disable_mask(mask) \
 	_xtos_ints_off(mask)
 
+static inline int arch_interrupt_register(int irq, void (*handler)(void *arg),
+					  void *arg)
+{
+	xthal_set_intclear(0x1 << irq);
+	_xtos_set_interrupt_handler_arg(irq, handler, arg);
+	return 0;
+}
```

Comment:
If my observation above is correct and my proposal is accepted, this will change, so presumably this will move to .c too.